### PR TITLE
CI: fix `pytest scripts`

### PR DESCRIPTION
**Pull request by MarcoGorelli**
seems that
```python
def pytest_addoption(parser):
    parser.addoption(
        "--strict-data-files",
        action="store_true",
        help="Unused. For compat with setup.cfg.",
    )
```
isn't needed anymore, and actually errors now:
```python
ValueError: option names {'--strict-data-files'} already added
```

### CODE REVIEW

File: pandas/compat/_optional.py [hunk 1]
```diff
@@ -36,7 +36,7 @@
     "pymysql": "1.0.2",
     "pyarrow": "7.0.0",
     "pyreadstat": "1.1.5",
-    "pytest": "7.0.0",
+    "pytest": "7.3.2",
```
Comment (MarcoGorelli) [inline pandas/compat/_optional.py:39]: @kathleenhang I'd have forgotten this one, but your script caught it, thanks 🙌

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @MarcoGorelli 

I guess we should backport?

**Comment by MarcoGorelli** [comment]
yeah probs

**Comment by mroeschke** [comment]
Why did pytest needed to be bumped? Technically we call out bumping these dev dependencies in the minor/major release whatsnews

**Comment by MarcoGorelli** [comment]
thanks @mroeschke , I'll make a PR to update this in the whatsnew

the reason is that the
```
def pytest_addoption(parser):
    parser.addoption(
        "--strict-data-files",
        action="store_true",
        help="Unused. For compat with setup.cfg.",
    )
```
options in `scripts/conftest.py` fail with a more modern pytest, so I figured it was easiest to just remove that and bump pytest

File: pandas/compat/_optional.py (hunk 1 above)
Comment (kathleenhang) [reply pandas/compat/_optional.py:39]: Thanks for the ping! It's really cool to see how it's helpful.

**Comment by mroeschke** [comment]
I think I am getting a related error in some PRs: https://github.com/pandas-dev/pandas/actions/runs/5383684555/jobs/9770736568?pr=53828

```
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --strict-data-files
  inifile: /home/runner/work/pandas/pandas/pyproject.toml
  rootdir: /home/runner/work/pandas/panda
```

Do you know why this is happening?

Also it appears `minversion = "7.0"` is still specified in pyproject.toml

**Comment by MarcoGorelli** [comment]
the logs show that that job is using:
```
pytest                         7.4.0 
```

My guess is that the issue might be in the `pytest-localserver` plugin?

**Pull request merged**